x-pack/plugins/enterprise_search/server/plugin.ts

```diff
@@ -6,30 +6,69 @@
 
 import { Observable } from 'rxjs';
 import { first } from 'rxjs/operators';
-import { Plugin, CoreSetup, PluginInitializerContext } from 'src/core/server';
+import {
+  Plugin,
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  SavedObjectsServiceStart,
+} from 'src/core/server';
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
 
 import { registerEnginesRoute } from './routes/app_search/engines';
+import { registerTelemetryRoute } from './routes/app_search/telemetry';
+import { registerTelemetryUsageCollector } from './collectors/app_search/telemetry';
+import { appSearchTelemetryType } from './saved_objects/app_search/telemetry';
+
+export interface PluginsSetup {
+  usageCollection?: UsageCollectionSetup;
+}
 
 export interface ServerConfigType {
   host?: string;
 }
 
 export class EnterpriseSearchPlugin implements Plugin {
   private config: Observable<ServerConfigType>;
+  private savedObjects?: SavedObjectsServiceStart;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.config = initializerContext.config.create<ServerConfigType>();
   }
 
-  public async setup({ http }: CoreSetup) {
+  public async setup(
+    { http, savedObjects, getStartServices }: CoreSetup,
+    { usageCollection }: PluginsSetup
+  ) {
     const router = http.createRouter();
     const config = await this.config.pipe(first()).toPromise();
     const dependencies = { router, config };
 
     registerEnginesRoute(dependencies);
+
+    /**
+     * Bootstrap the routes, saved objects, and collector for telemetry
+     */
+    registerTelemetryRoute({
```

Comment:
Should this throw an error if it gets an internalError response?

Reply:
Yeah, I think I need to write a new helper/commit to use Kibana's `logger` plugin and start logging caught errors to the server-side (only the Kibana admin has access to this data, but it should at least help them if they open a support ticket and we ask them to provide logs).